src/main/java/com/ziminer/xork/world/World.java

```diff
@@ -24,25 +26,63 @@ public static String coordToKey(int x, int y) {
     /**
      * Returns a world shaped like a grid, with
      * transitions between adjacent points.
-     *
+     * <p>
      * <p>The names of the tiles in the world are supplied
      * by {@link World#coordToKey(int, int)}:
-     *
+     * <p>
      * <p>0,0 is the bottom-left corner.
-     *
+     * <p>
      * E.g. for a 2 by 2 world:
-     *
+     * <p>
      * (0,1) - (1,1)
      * |         |
      * (0,0) - (1,0)
      *
-     * @param width The width of the world, in tiles.
+     * @param width  The width of the world, in tiles.
      * @param height The height of the world, in tiles.
      * @return The newly created {@link World}.
      */
     public static World newGridWorld(int width, int height) {
-        // TODO (roman, Sept 14 2016): Make this work.
-        throw new IllegalStateException("NOT WORKING YET.");
+        width--;
```

Comment:
Why go from (width - 1) to 0 instead of 0 to width (non-inclusive)? Makes the code look more complicated. You can just use:

for (int x = 0; x < width; ++x) etc.